src/pyfreeleh/row/serializers.py

```diff
@@ -0,0 +1,84 @@
+import abc
+from typing import Any, Dict, Generic, Type, TypeVar
+
+from pyfreeleh.providers.google.sheet.base import A1CellSelector
+from pyfreeleh.row import models
+from pyfreeleh.row.models import Model
+
+T = TypeVar("T", bound=Model)
+
+
+class Serializer(abc.ABC, Generic[T]):
+    def deserialize(self, data: Dict[str, Any]) -> T:
+        pass
+
+    def serialize(self, obj: T) -> Dict[str, Any]:
+        pass
+
+
+class FieldColumnMapper:
+    def __init__(self, model: Type[Model]):
+        self._model = model
+
+        self._col_name_by_field = self._get_col_name_mapping()
+        self._field_by_col_name = {v: k for (k, v) in self._col_name_by_field.items()}
+
+    def to_column(self, field: str) -> str:
+        return self._col_name_by_field[field]
+
+    def to_field(self, column: str) -> str:
+        return self._field_by_col_name[column]
+
+    def column(self, idx: int) -> str:
+        for i, column in enumerate(self._field_by_col_name.keys()):
+            if i == idx:
+                return column
+
+        raise ValueError
+
+    def field(self, idx: int) -> str:
+        for i, field in enumerate(self._field_by_col_name.values()):
+            if i == idx:
+                return field
+
+        raise ValueError
+
+    def col_by_field(self) -> Dict[str, str]:
+        return self._col_name_by_field
+
+    def field_by_col(self) -> Dict[str, str]:
+        return self._field_by_col_name
```

Comment:
Why need to use different names btw? Why not just use the same exact name?

Reply:
`Col` is the actual column name in the google sheet, e.g. `A`, `B`, `C`, etc

`field` here referring to the field name declared in the `Model`.

```
class A(Model):
  integer_field = IntegerField()
```

`integer_field` will be the `field`, and `B` will be its `col` name.

Reply:
No, I'm referring to why not callt he method `field_by_col_name()` instead.

Reply:
Done.